### PR TITLE
Eng 11366 windowed count

### PR DESCRIPTION
**Pull request by BillWhite**
Implement count(*) and count(EXPR) window functions.

https://issues.voltdb.com/browse/ENG-11366

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -497,6 +497,7 @@
     MergeReceiveExecutorTest
     TestGeneratedPlans
     TestRank
+    TestCount
```
Comment (wolffcm) [inline build.py]: This test just does windowed count, right?  So maybe make the name reflect this.

Comment (BillWhite) [inline build.py]: Fixed.  Also TestRank is renamed to TestWindowedRank.

File: src/ee/executors/windowfunctionexecutor.h [hunk 2]
```diff
@@ -34,27 +34,47 @@ class WindowFunctionExecutor: public AbstractExecutor {
...
         m_pmp(NULL),
         m_orderByKeySchema(NULL),
         m_partitionByKeySchema(NULL),
+        m_inputTable(NULL),
         m_inputSchema(NULL),
-        m_aggregateRow(NULL)
+        m_aggregateRow(NULL),
+        m_tableWindow(NULL)
         {
             dynamic_cast<WindowFunctionPlanNode *>(abstract_node)->collectOutputExpressions(m_outputColumnExpressions);
```
Comment (BillWhite) [inline src/ee/executors/windowfunctionexecutor.h]: I take it back.  These are references.  They need to be initialized in the constructor.

File: src/ee/executors/windowfunctionexecutor.h [hunk 3]
```diff
@@ -34,27 +34,47 @@ class WindowFunctionExecutor: public AbstractExecutor {
...
+     * depend on the order by peers of the row.  So, we need to
+     * scan the input forward to the next edge between order by
+     * groups.  Edges between partition by groups are a kind of
+     * edge between order by groups as well.
+     *
+     * This enum type gives the type of kinds of edges between groups
+     * of rows.
+     */
+    enum EdgeType {
+        InvalidEdgeType,         /** No Type. */
```
Comment (BillWhite) [inline src/ee/executors/windowfunctionexecutor.h]: Fixed.

File: src/ee/executors/windowfunctionexecutor.h [hunk 4]
```diff
@@ -148,27 +162,45 @@ class WindowFunctionExecutor: public AbstractExecutor {
...
+     * Call endGroup for each aggregate.  This will happen
+     * for each group and cannot be disabled.
+     */
+    void endGroupForAggs(TableWindow *window, EdgeType edgeType);
+    /**
+     * Insert the output tuple.
+     */
+    void insertOutputTuple();
 
     TupleSchema* constructSchemaFromExpressionVector(const AbstractPlanNode::OwningExpressionVector &exprs);
```
Comment (BillWhite) [inline src/ee/executors/windowfunctionexecutor.h]: Ok, I moved it, and renamed it createTupleSchema, which is what the rest of the similar factory functions are called.

File: tests/frontend/org/voltdb/regressionsuites/TestWindowedAggregateSuite.java [hunk 5]
```diff
@@ -693,6 +693,174 @@ public void testEng11029() throws Exception {
         }, vt);
     }
 
+    private void validateCount(Client client, String sql, long expected[][]) throws Exception {
```
Comment (BillWhite) [inline tests/frontend/org/voltdb/regressionsuites/TestWindowedAggregateSuite.java]: Actually, I ran the memcheck tests and they all passed.  I'm writing tests with more data types and partitioned tables.  But as far as data types go, the WindowAggregate objects, which do all the calculation, contain only NValues with integers in them.  We don't actually save any strings there.  Consequently I'm not sure what a good set of tests would be.   A white box test would be one which tried to create an NValue string which didn't get collected, or which got garbage collected some how before it was used.  I'm not sure how that could happen.  But I'm writing the tests as well as I can.

**Pull request merged**